providers/storetheindex/findproviders.go

```diff
@@ -56,9 +62,11 @@ func (c *client) FindProviders(ctx context.Context, mh multihash.Multihash) ([]p
 	if len(parsedResponse.MultihashResults) != 1 {
 		return nil, fmt.Errorf("unexpected number of responses")
 	}
-	result := make([]peer.AddrInfo, len(parsedResponse.MultihashResults[0].ProviderResults))
+	result := make([]peer.AddrInfo, 0, len(parsedResponse.MultihashResults[0].ProviderResults))
 	for _, m := range parsedResponse.MultihashResults[0].ProviderResults {
-		result = append(result, m.Provider)
+		if bytes.HasPrefix(m.Metadata, bitswapPrefix) {
```

Comment:
How do you know that the metadata will have this prefix?

Reply:
The spec we settled on says that supported protocols should be uvarints in sorted order. bitswap is the smallest defined transport, so if it's supported it'll be at the beginning.

Reply:
Interesting so the protocols are not in user-specified order? Is this mostly because the current indexer advertisements are not for lists of transports but instead separate advertisements per transport? If so is the a short or longer lived expectation (not that this is relevant to merging this PR specifically, but trying to understand how it'll operate with other systems that allow suggested prioritization/ordering).

Reply:
this was a constraint on extending from 1 protocol to an array of them in a backwards compatible way with the current wire protocol.

Once we switch to something compatible with the records being discussed in reframe we'll make a switch once in what we ask from providers and will remove this expectation.